### ISSUE TITLE
Analysis of example models of MSL v4.0.0-beta.1 failing a pedantic translation

### ISSUE THREAD

**Issue by beutlich** [modelica/ModelicaStandardLibrary#3435]
The following models/blocks with StopTime annotation fail translation if pedantic=true:

- [x] Modelica.Blocks.Examples.PID_Controller
- [x] Modelica.Clocked.Examples.Systems.EngineThrottleControl
- [x] Modelica.Clocked.Examples.Elementary.ClockSignals.RotationalSample
- [x] Modelica.Clocked.Examples.Elementary.ClockSignals.LogicalSample
- [x] Modelica.Electrical.Spice3.Examples.Oscillator
- [x] Modelica.Electrical.Spice3.Examples.CoupledInductors
- [x] Modelica.Magnetic.FluxTubes.Examples.Hysteresis.HysteresisModelComparison
- [x] Modelica.Magnetic.FluxTubes.Examples.Hysteresis.SinglePhaseTransformerWithHysteresis1
- [x] Modelica.Magnetic.FluxTubes.Examples.Hysteresis.SinglePhaseTransformerWithHysteresis2
- [x] Modelica.Magnetic.FluxTubes.Examples.Hysteresis.ThreePhaseTransformerWithRectifier
- [x] Modelica.Magnetic.QuasiStatic.FluxTubes.Examples.Leakage.CylinderLeakage
- [x] Modelica.Magnetic.QuasiStatic.FluxTubes.Examples.Leakage.GeneralLeakage
- [ ] Modelica.Fluid.Examples.HeatingSystem
- [ ] Modelica.Fluid.Examples.AST_BatchPlant.Test.OneTank
- [ ] Modelica.Fluid.Examples.NonCircularPipes
- [x] Modelica.Fluid.Examples.TraceSubstances.RoomCO2WithControls
- [x] Modelica.Fluid.Examples.InverseParameterization
- [ ] Modelica.Fluid.Examples.Explanatory.MeasuringTemperature

**Comment by beutlich** [comment]
Hysterisis models are addressed by #3427.

**Comment by beutlich** [comment]
Spice3 models are addressed by #3438.

**Comment by casella** [comment]
@beutlich, I just checked all the Modelica.Fluid examples on branch v4.0.0-beta.1 using Dymola 2020x and pedantic mode, but I don't get any error or warning. Can you please report which errors do you get?

**Comment by beutlich** [comment]
Here the warnings:

```
Translation of Modelica.Fluid.Examples.HeatingSystem:

Some variables are iteration variables of the initialization problem:
but they are not given any explicit start values. Zero will be used.

Iteration variables:
der(tank.V)
```

```
Translation of Modelica.Fluid.Examples.AST_BatchPlant.Test.OneTank:

The initial conditions are not fully specified.
Dymola has selected default initial conditions.

Assuming fixed start value for the continuous states:
tank.level(start = tank.level_start)
tank.medium.T(start = tank.T_start)
```

```
Translation of Modelica.Fluid.Examples.NonCircularPipes:

Redundant consistent initial conditions.

Removed the following equations which are redundant and consistent: 
circular_pipe.port_a.p = circular_pipe.mediums[1].p;
annulus_pipe.port_a.p = annulus_pipe.mediums[1].p;

This required evaluation of 
the interactive parameters 
boundary.p(start = 1000000.0)

The redundancies involve
initial equations: 
circular_pipe.mediums[1].p = 1000000.0;
annulus_pipe.mediums[1].p = 1000000.0;
```

```
Translation of Modelica.Fluid.Examples.TraceSubstances.RoomCO2WithControls:

The initial conditions are not fully specified.
Dymola has selected default initial conditions.

Assuming fixed start value for the continuous states:
PID.I.y(start = PID.I.y_start)
```

```
Translation of Modelica.Fluid.Examples.Explanatory.MeasuringTemperature:

Some variables are iteration variables of the initialization problem:
but they are not given any explicit start values. Zero will be used.

Iteration variables:
der(openTankCold3.m)
der(openTankHot1.m)
der(openTankHot2.m)
```

I can no longer reproduce Modelica.Fluid.Examples.InverseParameterization.

**Comment by beutlich** [comment]
@AHaumer @christiankral Can you please have a look at Modelica.Blocks.Examples.PID_Controller. It is not a big deal to set this initial condition (somewhere hidden in the component), the actual trouble is, if PID is missing such an option generally.

**Comment by christiankral** [comment]
> @AHaumer @christiankral Can you please have a look at Modelica.Blocks.Examples.PID_Controller. It is not a big deal to set this initial condition (somewhere hidden in the component), the actual trouble is, if PID is missing such an option generally.

@beutlich So this means that we actually need a new ticket to consistently set the initial conditions in each case of controller configurations (P, PI, PD, PID).

**Comment by beutlich** [comment]
Yes, that is the question I was asking myself.

**Comment by beutlich** [comment]
Removing milestone after discussion stopped.

**Comment by beutlich** [comment]
Modelica.Fluid.Examples.TraceSubstances.RoomCO2WithControls was fixed en passant by a0b60a045ba45bdf0bd6e11573a791cca958d3ac.

**Comment by beutlich** [comment]
Closing after MSL v4.0.0 was released. It is a "WontFix" for the four Fluid examples that still raise initialization warnings in pedantic mode.

- [ ] Modelica.Fluid.Examples.HeatingSystem
- [ ] Modelica.Fluid.Examples.AST_BatchPlant.Test.OneTank
- [ ] Modelica.Fluid.Examples.NonCircularPipes
- [ ] Modelica.Fluid.Examples.Explanatory.MeasuringTemperature
